### PR TITLE
Fix dark liquids

### LINKED ISSUE #6597: Liquids emitting little light are dark
The problem appears only with smooth lighting disabled.

Here, `light_source` is 0:
screenshot_20171107_005730

And here, `light_source` is 1:
screenshot_20171107_005716

The problem is that at [content_mapblock.cpp:397](https://github.com/minetest/minetest/blob/master/src/content_mapblock.cpp#L397) light level for light-emitting liquid is overridden by its `light_source` value even if the latter is smaller. (Note that the problem is there since d8edd3d0c8 [yes that’s 5 years ago]; it might not be noticed because no one actually uses low `light_source` values).

### PR DESCRIPTION
**Pull request by numberZero**
Fixes #6597

### CODE REVIEW

File: src/content_mapblock.h [hunk 1]
```diff
@@ -25,8 +25,25 @@ with this program; if not, write to the Free Software Foundation, Inc.,
...
+	union {
+		u16 light;
+		struct {
+			u8 lightA;
+			u8 lightB;
+		};
+	};
+
+	LightPair() = default;
+	explicit LightPair(u16 value) : light(value) {}
```
Comment (sfan5) [inline src/content_mapblock.h]: this is endian-dependent, probably not what you want
also the compiler might insert padding between `lightA` and `lightB` so the union thing is not even guaranteed to work

Comment (numberZero) [reply src/content_mapblock.h]: > this is endian-dependent

Not a problem as long as you don’t mix bitwise operations with union member access. It is not guaranteed that `lightA` stores any specific bank, but it *is* guaranteed that it stores the same bank everywhere.

> the compiler might insert padding

Alignment requirements for a type can’t exceed its size, which is 1 byte in this case.

> the union thing is not even guaranteed to work

My mistake; according to cppreference.com such thing **is** guaranteed to work in C, but not in C++:
> If the member used to access the contents of a union is not the same as the member last used to store a value, the object representation of the value that was stored is reinterpreted as an object representation of the new type (this is known as type punning).

> it's undefined behavior to read from the member of the union that wasn't most recently written

**Comment by numberZero** [comment]
@sfan5 All checks except of LINT have passed. But even LINT didn’t failed but “errored” instead.

**Comment by sfan5** [comment]
LGTM :+1:

**Pull request merged**